state_test.go

```diff
@@ -1729,15 +1760,16 @@ func TestMemberlist_GossipToDead(t *testing.T) {
 
 	// Should gossip to m2 because its state has changed within GossipToTheDeadTime
 	m1.nodeMap[addr2.String()].StateChange = time.Now().Add(-20 * time.Millisecond)
-	m1.gossip()
 
-	for i := 0; i < 2; i++ {
-		select {
-		case <-ch:
-		case <-time.After(50 * time.Millisecond):
-			t.Fatalf("timeout")
+	retry(t, 5, 10*time.Millisecond, func(failf func(string, ...interface{})) {
+		m1.gossip()
+
+		time.Sleep(3 * time.Millisecond)
```

Comment:
Are we expecting 1 message per Millisecond?

If so couldn't we sleep for `3333 * time.Microsecond` or something with a little extra to increase reliability of the tests?

I have very little context here so I could be wrong.

Reply:
No we are expecting them all to be delivered, this is just giving go scheduler and network stack a bit of space otherwise it would likely fail every single time on the first retry. Could remove it and let it always fail and be retried but seemed better to leave it with a chance of passing!